impl/client/src/main/javascript/web/dojo/pentaho/common/PageControl.js

```diff
@@ -30,161 +30,185 @@
-          return;
-        }
-        this.setPageNumber(this.pageNumberInput.get("value"));
-      },
-
-      _pageTotalMouseDown: function(e) {
-        e.preventDefault();
-      },
-
-      _updateInternalPageNumber: function(pageNumber) {
-        if (this.changePageNumberCallback) {
-          this.changePageNumberCallback(pageNumber);
-        }
-        this.pageNumber = pageNumber;
-        this.pageNumberInput.set("value", this.pageNumber, false);
-        this._updatePageButtonState();
-      },
-
-      _pageNumberKeyUp: function(event) {
-        switch (event.keyCode) {
-          case keys.ENTER:
-            this._changePageNumber();
-            break;
-          case keys.ESCAPE:
-            this._resetPageNumber();
-            break;
-          default:
-            // We don't care about any other key
-            break;
-        }
-      },
-
-      _resetPageNumber: function() {
-        this.pageNumberInput.set("value", this.pageNumber);
-        this._updatePageButtonState();
-      },
-
-      // Change to the previous page
-      _prevPage: function() {
-        if (this.pagePrevDisabled) {
-          return;
-        }
-        this.setPageNumber(this.pageNumber - 1);
-      },
-
-      // Change to the next page
-      _nextPage: function() {
-        if (this.pageNextDisabled) {
-          return;
-        }
-        this.setPageNumber(this.pageNumber + 1);
-      }
-    });
-  }
+
+define(["dojo/_base/declare", "dojo/on", "dojo/query", "dojo/_base/lang", "dijit/_WidgetBase", "dijit/_TemplatedMixin", "dijit/_WidgetsInTemplateMixin",
+        "dijit/form/NumberTextBox", "dojo/dom-class", "dojo/keys", "common-ui/util/_a11y"],
+    function (declare, on, query, lang, _WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin, NumberTextBox, domClass, keys, a11yUtil) {
+        return declare("pentaho.common.PageControl", [_WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin], {
+
+            // Template for this widget
+            templateString: "<div class='pc_pageControlContainer' data-dojo-attach-point='containerNode'><div class='pc_pagePrev' data-dojo-attach-point='prevPageControl'></div><div class='pc_pageNext' data-dojo-attach-point='nextPageControl'></div><div data-dojo-attach-point='pageNumberInput' data-dojo-type='dijit.form.NumberTextBox' class='pc_pageNumberInput' constraints='{min:1}'></div><div class='pc_pageTotal contrast-color' data-dojo-attach-point='pageTotalSpan'>/ ##</div></div>",
+
+            // Function for retrieving localized strings
+            getLocaleString: null,
+
+            // Callback to set page number
+            changePageNumberCallback: null,
+
+            // Current page number
+            pageNumber: 1,
+
+            // Total pages in this report
+            pageTotal: 1,
+
+            // Called once per instance
+            postCreate: function () {
+                this._resetPageNumber();
+                this.setPageCount(this.pageTotal);
+                this.own(
```

Comment:
Good move to replace the deprecated `connect` by `on` and to start using `own` to reduce memory leakage.